back/engines/commercial/report_builder/app/services/report_builder/queries/projects.rb

```diff
@@ -0,0 +1,67 @@
+module ReportBuilder
+  class Queries::Projects < ReportBuilder::Queries::Base
+    def run_query(start_at: nil, end_at: nil, **_other_props)
+      start_date, end_date = TimeBoundariesParser.new(start_at, end_at).parse
+
+      projects_that_have_phases_ids = Phase.select(:project_id)
+
+      far_future_date = Time.zone.today + 10.years
+
+      project_periods = Phase
+        .group(:project_id)
+        .select("project_id, min(start_at) as start_at, max(start_at) as last_phase_start_at, max(coalesce(end_at, '#{far_future_date}'::DATE)) as end_at")
+        .to_a
+
+      non_overlapping_project_period_ids = project_periods
+        .select do |period|
+          period.start_at >= end_date || period.end_at <= start_date
+        end
```

Comment:
To gracefully deal with missing `end_at` dates, you can use something like:
```suggestion
      overlapping_project_period_ids = project_periods.select do |period|
        project_period = (period.start_at..period.end_at)
        project_period.overlaps?(start_date..end_date)
      end
```